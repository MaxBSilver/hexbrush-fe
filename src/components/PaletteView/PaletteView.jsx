import React from 'react';
import Palette from '../Palette/Palette';

const PaletteView = props => {
	const { palettes } = props;
	return <output class="PaletteView">{palettes.map(palette => <Palette key={palette.id} {...palette} />)}</output>;
};

export default PaletteView;
