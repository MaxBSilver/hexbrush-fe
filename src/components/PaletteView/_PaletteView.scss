.PaletteView {
	display: grid;
	padding: 50px;
	grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
	grid-gap: 50px;
	justify-items: center;
}

.PaletteView-edit-form {
	display: flex;
}

.PaletteView-button {
	@include input-style;
	width: 250px;
	margin-left: 15px;

	&:first-child {
		margin-left: 0;
	}
}

.PaletteView-edit-button {
	width: 100px;
}
