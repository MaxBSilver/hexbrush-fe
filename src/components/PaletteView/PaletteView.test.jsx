import React from 'react';
import PaletteView from '../PaletteView/PaletteView';
import { shallow } from 'enzyme';

const mockPalettes = [{
	color_1: '#899e01',
	color_2: '#ff21c3',
	color_3: '#88b712',
	color_4: '#f7c574',
	color_5: '#6520cc'
}];

describe('PaletteView', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<PaletteView palettes={mockPalettes} />);
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
