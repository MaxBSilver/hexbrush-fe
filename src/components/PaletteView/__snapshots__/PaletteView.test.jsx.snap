// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteView should match the snapshot 1`] = `ShallowWrapper {}`;
